pensieve/tests/test_config.py

```diff
@@ -0,0 +1,131 @@
+from textwrap import dedent
+
+import toml
+import pytest
+
+from pensieve import config
+
+
+class TestAnalysisSpec:
+    def test_trivial_configuration(self, experiments):
+        spec = config.AnalysisSpec.from_dict({})
+        assert isinstance(spec, config.AnalysisSpec)
+        cfg = spec.resolve(experiments[0])
+        assert isinstance(cfg, config.AnalysisConfiguration)
+        # There should be some default metrics
+        assert len(cfg.metrics.weekly)
+
+    def test_scalar_metrics_throws_exception(self):
+        config_str = dedent(
+            """
+            [metrics]
+            weekly = "my_cool_metric"
+            """
+        )
+        with pytest.raises(ValueError):
+            config.AnalysisSpec.from_dict(toml.loads(config_str))
+
+    def test_template_expansion(self, experiments):
+        config_str = dedent(
+            """
+            [metrics]
+            weekly = ["my_cool_metric"]
+
+            [metrics.my_cool_metric]
+            data_source = "main"
+            select_expression = "{{agg_histogram_mean('payload.content.my_cool_histogram')}}"
```

Comment:
This is something we'll have to document very clearly. My feeling is that a common error when writing these config files might be that people forget it's a jinja template.

Alternatively, I wonder if it makes sense to not enforce writing this as a template in the config, but instead check if the "raw" value is a valid SQL query or if it's valid SQL after handling it as a template.

Reply:
Maybe, once we have a validator for config files this won't be much of an issue.

Reply:
It's nice that Jinja syntax doesn't really conflict with BigQuery syntax; there aren't a lot of curly braces in SQL. So I think the risk is that people won't realize that they can use the helper methods. Passing valid SQL through Jinja should be a no-op.

I think the helpers are going to be used for even straightforward expressions, so I think they'll show up in the examples that people use; I agree it'll be important to document them.

I'm hoping the CLI will have a command for printing the available helpers.